platform/core/crates/update_notification/src/lib.rs

```diff
@@ -0,0 +1,113 @@
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::sync::Arc;
+
+#[derive(Eq, PartialEq, Copy, Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum Seen {
+    Not,
+    Push,
+    All,
+}
+
+#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LastNotification {
+    pub id: u32,
+    pub state: Seen,
+}
+
+pub trait UpdateNotificationStore: Send + Sync {
+    fn get_last_notification(&self) -> Option<LastNotification>;
+    fn set_last_notification(&self, last_notification: LastNotification);
+}
+
+pub struct UpdateNotification {
+    pub id: u32,
+    store: Arc<dyn UpdateNotificationStore>,
+}
+
+impl UpdateNotification {
+    pub fn new(id: u32, store: Arc<dyn UpdateNotificationStore>) -> Self {
+        Self {
+            id,
+            store: store.clone(),
+        }
+    }
+
+    fn get_last_notification(&self) -> LastNotification {
+        self.store
+            .get_last_notification()
+            .unwrap_or(LastNotification {
+                id: 0,
+                state: Seen::Not,
+            })
+    }
+
+    fn show<F: Fn(Seen) -> bool>(&self, show_for_last_state: F) -> bool {
+        let last_notification = self.get_last_notification();
+        if self.id == 0 || self.id < last_notification.id {
+            return false;
+        }
+        if self.id > last_notification.id {
+            return true;
+        }
+        show_for_last_state(last_notification.state)
+    }
+
+    pub fn handle_first_run(&mut self) {
+        self.handle_in_app_seen();
+    }
+
+    pub fn handle_startup(&mut self) {
+        self.handle_push_seen();
+    }
+
+    pub fn show_in_app(&self) -> bool {
```

Comment:
This feels like the name it should be a question, like `should_show_in_app`, same for `show_push`.

Reply:
Good point, I'll change it!

Reply:
Done - went a bit further and also renamed CoreRequest's `GetShow*` to `ShouldShow*`.